dart/lib/src/protocol/sentry_response.dart

```diff
@@ -54,7 +54,10 @@ class SentryResponse {
   factory SentryResponse.fromJson(Map<String, dynamic> json) {
     return SentryResponse(
       url: json['url'],
-      headers: json['headers'],
+      headers: json.containsKey('headers')
+          ? (json['headers'] as Map<String, dynamic>)
+              .map((key, value) => MapEntry(key, value as String))
+          : null,
```

Comment:
We just do `headers: json['headers']`, any reason why the defensive approach? only the SDK calls `SentryResponse.fromJson` from anyway, `value as String` cal also fails if `value` isn't a `String`.

Reply:
assigning json['headers'] would fail because it's `Map<String, dynamic>` and the target is `Map<String, string>`. I've changed to `Map.from()` though - should work just fine here as well.

The same logic for other similar cases [was historically implemented in this test code](https://github.com/getsentry/sentry-dart/pull/1095/files#diff-3825cfe5b084ce396d606318e82b63a3f7b736e7924b34836fe115adeb1bbb7dR49-R73), which, AFAICT, is also currently the only place executing `fromJSON` for these types.

Reply:
Yeah that makes sense, the reason was that the test deserializes from a file, but Dart won't do it, its always created by the SDK or the user do it manually, in this case, they don't use the `fromJson` but rather the normal ctor respecting the data structure type, `Map.from()` should do.